Programs/python.c

```diff
@@ -53,9 +60,30 @@ main(int argc, char **argv)
     fedisableexcept(FE_OVERFLOW);
 #endif
 
+    /* UTF-8 mode */
+    char *ctype = setlocale(LC_CTYPE, "");
+    if (ctype != NULL) {
+        ctype = _PyMem_RawStrdup(ctype);
+        if (!ctype) {
+            PyMem_RawFree(ctype);
+            fatal_error("out of memory");
+        }
+        if (strcmp(ctype, "C") == 0) {
+            /* The POSIX locale enables the UTF-8 mode */
+            Py_UTF8Mode = 1;
+        }
+        setlocale(LC_CTYPE, ctype);
+        PyMem_RawFree(ctype);
```

Comment:
What is this setlocale() for?

Reply:
The line 64 changes the locale to the user LC_CTYPE locale. This line restores the locale to its previous value.

Reply:
Shouldn't previous value be got by `setlocale(LC_CTYPE, NULL)` instead of `setlocale(LC_CTYPE, "")`?

Reply:
Oh, you're right. I fixed it in a new commit.